### PR TITLE
Improved error handling all around.

### PR DESCRIPTION
**Pull request by masotime**
1. [jshint] removed unused variables, added / removed semicolons, use single quotes consistently
2. Where there is a callback, always return an error if it exists (not sure why it was "null" previously most of the time"
3. Wrap dust rendering in a try catch so errors are handled in callback rather than thrown.
4. Additional tests for all callback errors.

### CODE REVIEW

File: bundler/dust.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ var spud = require('spud');
 var freshy = require('freshy');
 var loopalo = require('../lib/loopalo');
 //import dust here to get around issue in freshy where freshy.freshy doesn't properly check for an existing copy
-var doost = require('dustjs-linkedin');
+require('dustjs-linkedin');
```
Comment (aredridel) [inline bundler/dust.js]: This should get fixed in freshy then!

Comment (totherik) [inline bundler/dust.js]: Not sure when that comment was written, but it should support that feature.

Comment (aredridel) [inline bundler/dust.js]: Definitely. My last use of freshy supported it well.

**Comment by pvenkatakrishnan** [comment]
:+1:

File: bundler/dust.js (hunk 1 above)
Comment (masotime) [inline bundler/dust.js]: Use freshy to load `dustjs-linkedin` instead of using the require() call?

Comment (aredridel) [inline bundler/dust.js]: I think that first require just isn't needed at all, now. `freshy` should work fine even when the library hasn't been loaded yet.

**Pull request merged**